### PR TITLE
[4.0] Validation in Mass Mail Users

### PR DESCRIPTION
**Pull request by rjharishabh**
Pull Request for Issue #33450  

### Summary of Changes
form-validation added

### Testing Instructions
`Users` > `Mass Mail Users` > `Click Send Email`
Apply PR
npm i
Repeat and See the changes

### Actual result BEFORE applying this Pull Request
before-mass

### Expected result AFTER applying this Pull Request
user-mail-after

### Documentation Changes Required
None

### CODE REVIEW

**Comment by YatharthVyas** [comment]
This If-Else block is redundant @rjharishabh 
Please have a look

```
if (pressbutton === 'mail.cancel') {
    Joomla.submitform(pressbutton);
    return;
} else {
    Joomla.submitform(pressbutton);
}
```
      
https://github.com/joomla/joomla-cms/blob/4c45f7929495e5f2dfc4defcf9849036689be110/build/media_source/com_users/js/admin-users-mail.es6.js#L14

File: build/media_source/com_users/js/admin-users-mail.es6.js [hunk 1]
```diff
@@ -14,18 +14,6 @@
       if (pressbutton === 'mail.cancel') {
```
Comment (YatharthVyas) [inline build/media_source/com_users/js/admin-users-mail.es6.js]: This If-Else block is redundant @rjharishabh 
Please have a look

```
if (pressbutton === 'mail.cancel') {
    Joomla.submitform(pressbutton);
    return;
} else {
    Joomla.submitform(pressbutton);
}
```

Comment (rjharishabh) [reply build/media_source/com_users/js/admin-users-mail.es6.js]: I got it Thanks

**Comment by Quy** [comment]
Is there a need for the JavaScript file as validation is no longer done in it?

File: administrator/components/com_users/tmpl/mail/default.php [hunk 2]
```diff
@@ -14,18 +14,15 @@
...
 use Joomla\CMS\Router\Route;
 
-Text::script('COM_USERS_MAIL_PLEASE_FILL_IN_THE_SUBJECT', true);
-Text::script('COM_USERS_MAIL_PLEASE_SELECT_A_GROUP', true);
-Text::script('COM_USERS_MAIL_PLEASE_FILL_IN_THE_MESSAGE', true);
-
 /** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
 $wa = $this->document->getWebAssetManager();
-$wa->useScript('com_users.admin-users-mail');
+$wa->useScript('com_users.admin-users-mail')
```
Comment (Quy) [inline administrator/components/com_users/tmpl/mail/default.php]: Please remove.

**Comment by richard67** [comment]
I've updated the branch to latest 4.0-dev to get rid of the unrelated javascript-cs error in drone.

**Comment by YatharthVyas** [comment]
I tested your PR and it works as mentioned. Great Work @rjharishabh!

However, I noticed that a very similar form exists for Private Messages where The Subject and the Message is validated. Here, the validation prompt is displayed in a slightly different style with the list of all the fields having an error:

image

An improvement suggestion: such an implementation can provide better UX feedback to the user and it will also maintain consistency among the mail/message component forms. What's your opinion?

**Comment by rjharishabh** [comment]
I will look into it

**Comment by Kostelano** [comment]
I agree with @YatharthVyas. However, the PR works well.

**Comment by ceford** [comment]
I have tested this item :red_circle: unsuccessfully on 6af1e7a1106a024fff78ce55c2a46c299335312c<br /><br />The individual fields work but the form submits with the subject and message fields both empty. Is that was what the javascript file was for?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33485">issues.joomla.org/tracker/joomla-cms/33485</a>.</sub>

**Pull request closed without merging**